jax/_src/random.py

```diff
@@ -36,6 +36,15 @@
 from jax._src.util import prod
 
 
+IntOrFloat = Union[float, int, np.integer, np.floating]
+IntFloatOrArray = Union[jnp.ndarray, IntOrFloat]
+IntOrArray = Union[jnp.ndarray, int, np.integer]
+# TODO: Import or define these to match
+# https://github.com/numpy/numpy/blob/main/numpy/typing/_dtype_like.py.
+DTypeLikeInt = Any
+DTypeLikeFloat = Any
```

Comment:
I replaced some of the `np.dtype` annotations with these aliases currently set to `Any`.  Numpy added these types internally six weeks ago, which cover all the bases.  The most common thing to pass as dtype is probably `np.float32` or `'f'` and neither of those are even an `np.dtype`.